### ISSUE TITLE
CI_STEP_NUMBER is set as 0

### ISSUE THREAD

**Issue by eliasscosta** [woodpecker-ci/woodpecker#3163]
### Component

server, agent

### Describe the bug

Variable CI_STEP_NUMBER is set as 0.
I'm using 2.1.1 with Kubernetes backend; all pods are created with 0. 

### System Info

```shell
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"2.1.1"}
```

### Additional context

![image](https://github.com/woodpecker-ci/woodpecker/assets/22245125/12c83664-6d1d-4a5f-a1f2-9c608bdf69ca)

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by 6543** [comment]
Should changed in v2.2.2 ... is it still wrong?

**Comment by 6543** [comment]
The dag related stuff changed things in this

**Comment by 6543** [comment]
I'll have a deeper look

**Comment by 6543** [comment]
yes looked into it ... I'm currently working on it - it's not that easy. I have to change some internal functions etc.

**Comment by 6543** [comment]
same as https://github.com/woodpecker-ci/woodpecker/issues/1760
